pkg/addons/irsa_helper.go

```diff
@@ -46,20 +48,20 @@ func (h *irsaHelper) IsSupported() (bool, error) {
 }
 
 // CreateOrUpdate creates IRSA for the specified IAM service accounts or updates it
-func (h *irsaHelper) CreateOrUpdate(sa *api.ClusterIAMServiceAccount) error {
+func (h *irsaHelper) CreateOrUpdate(ctx context.Context, sa *api.ClusterIAMServiceAccount) error {
 	serviceAccounts := []*api.ClusterIAMServiceAccount{sa}
 	name := makeIAMServiceAccountStackName(h.clusterName, sa.Namespace, sa.Name)
-	stack, err := h.stackManager.DescribeStack(&manager.Stack{StackName: &name})
+	stack, err := h.stackManager.DescribeStack(ctx, &manager.Stack{StackName: &name})
 	if err != nil {
-		if awsError, ok := errors.Unwrap(errors.Unwrap(err)).(awserr.Error); !ok || ok &&
-			awsError.Code() != "ValidationError" {
+		if awsError, ok := errors.Unwrap(errors.Unwrap(err)).(*smithy.OperationError); !ok || ok &&
+			!strings.Contains(awsError.Error(), "ValidationError") {
```

Comment:
we have an `isValidationError` somewhere for that in here. Might be worth exporting that and using it here.

Reply:
:thinking: ill take a look!

Reply:
I can't re-use that as its for other v1 API calls, but ill add a new one for v2.